docs/developer/v1alpha1-compared-to-v1alpha2.md

```diff
@@ -0,0 +1,66 @@
+# Cluster API v1alpha1 compared to v1alpha2
+
+## Providers
+
+### v1alpha1
+
+Providers in v1alpha1 wrap behavior specific to an environment to create the infrastructure and bootstrap instances into
+Kubernetes nodes. Examples of environments that have integrated with Cluster API v1alpha1 include, aws, gcp, openstack,
+azure, vsphere and others. The provider vendors in Cluster API's controllers, registers their actuators with the
+controllers and runs a single manager (with the Cluster API controllers) to complete a Cluster API management cluster.
+
+### v1alpha2
+
+v1alpha2 introduces new providers. Cluster API's controllers are no longer vendored by providers. Instead, Cluster API 
+offers its own independent controllers that are responsible for the core types:
+
+* Cluster
+* Machine
+* MachineSet
+* MachineDeployment
+* [plus a few more](../../pkg/apis/cluster/v1alpha2)
+
+To complete a management cluster in v1alpha2, the Cluster API controllers need support from two other providers. A Bootstrap
+Provider and an Infrastructure provider.
+
+The Bootstrap provider is responsible for running a controller that generates data necessary to bootstrap an instance into a 
+Kubernetes node (cloud-init, bash, etc).
+
+The Infrastructure provider is responsible for generating actual infrastructure (networking, load balancers, instances, 
+etc) for a particular environment that can consume bootstrap data to turn the infrastructure into a Kubernetes cluster.
+
+This means that in order to have a complete management cluster that is ready to build clusters you will need three managers.
+
+v1alpha1 "providers" have become Infrastructure providers. Cluster API has moved from being a library to vendor to its
```

Comment:
For v1alpha1, it was half library, half its own proper controllers.

Reply:
i don't think it's necessary to get into the details of exactly what cluster-api was, thinking about cluster api as a library seems sufficient to me for the purpose of this document.

Do you think there's a benefit to getting into the details about how capi was half library half independent controllers? Like, does omitting that detail really harm the understanding?

Reply:
I think this is fine, thanks.